src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -314,19 +338,99 @@ private void updateFlightState(String comment, String sql, FlightContext flightC
               new NamedParameterPreparedStatement(connection, sql)) {
 
         startTransaction(connection);
-        statement.setString("status", flightContext.getFlightStatus().name());
-        statement.setString("flightId", flightContext.getFlightId());
+        statement.setString("status", status);
+        statement.setString("flightId", flightId);
+        statement.getPreparedStatement().executeUpdate();
+        commitTransaction(connection);
+        break;
+      } catch (SQLException ex) {
+        if (!retrySqlException(ex)) {
+          throw new DatabaseOperationException(
+              "Failed to update status to " + status + " for flight: " + flightId, ex);
+        }
+      }
+      retryWait(comment, flightId);
+    }
+  }
+
+  /**
+   * This method is used during recovery of an obsolete stairway instance. It "disowns" all of the
+   * flights that were owned by that stairway and puts them in the READY state. It then removes the
+   * obsolete stairway instance from the stairway_instance table.
+   *
+   * @param stairwayId the id of the, presumably deleted, stairway instance
+   * @return list of the flight ids that we disowned.
+   * @throws DatabaseOperationException on database error
+   * @throws InterruptedException interruption of the retry loop
+   */
+  void disownRecovery(String stairwayId) throws DatabaseOperationException, InterruptedException {
+    final String sql =
+        "UPDATE "
```

Comment:
Do any of these need `@Transactional`-type annotations?

Reply:
I think no, because he's doing "manual" transaction start/end in the method.